File: leetcode_py/p2240.py
class Solution:
    def waysToBuyPensPencils(self, total: int, cost1: int, cost2: int) -> int:
        if cost1 < cost2:
            return self.waysToBuyPensPencils(total, cost2, cost1)
        res = cnt = 0
        while total >= cost1 * cnt:
            res += (total - cost1 * cnt) // cost2 + 1
            cnt += 1
        return res

File: leetcode_C++/p2240.cpp
/*
@Author: mrxhar
@Date:   2023-09-01 09:54:40
@Email:  1337904145@qq.com
@Blog:   https://leetcode.cn/problems/number-of-ways-to-buy-pens-and-pencils/?envType=daily-question&envId=2023-09-01
*/
#include <iostream>

using namespace std;

class Solution
{
public:
    long long waysToBuyPensPencils(int total, int cost1, int cost2)
    {
        if (cost1 < cost2)
        {
            return waysToBuyPensPencils(total, cost2, cost1);
        }
        long res = 0, cnt = 0;
        while (total - cnt * cost1 >= 0)
        {
            res += (total - cnt * cost1) / cost2 + 1;
            cnt++;
        }
        return res;
    }
};